### ISSUE TITLE
Checkerboard artefacts with hdr-compute-peak=auto

### ISSUE THREAD

**Issue by badgerkin** [mpv-player/mpv#13051]
### Important Information

- mpv version: `v0.37.0-70-g562450f59e`
- Linux Distribution and Version: `Ubuntu 23.10 mantic`
- Source of the mpv binary: Debian package built from source with `mpv-build`
- If known which version of mpv introduced the problem: Unknown
- Window Manager and version: `gdm3 45.beta`
- GPU driver and version: `i915`  from Linux `6.5.0-14-generic` | `vainfo: Intel iHD driver for Intel(R) Gen Graphics - 23.2.3`

### Reproduction steps

Play [Exodus_HDR.mp4](https://0x0.st/H3bq.mp4) with the following options:

```shell
mpv --no-config --hdr-compute-peak=auto --log-file=Exodus_HDR_mpv_compute_peak_auto.log Exodus_HDR.mp4
```

```shell
mpv --no-config --hdr-compute-peak=no --log-file=Exodus_HDR_mpv_compute_peak_no.log Exodus_HDR.mp4
```

### Expected behavior

Video plays without obvious artefacts with the default option `hdr-compute-peak=auto`  set. HDR peak computation should be enabled only if supported by the driver.

### Actual behavior

With `--hdr-compute-peak=auto` set, I observe artefacts like these when playing some files:
![exodus_artefact](https://github.com/mpv-player/mpv/assets/50284661/da3049ac-ca68-4cac-854b-42a7192123cf)

Video playback appears normal with `--hdr-compute-peak=no `.

### Log file

[Exodus_HDR_mpv_compute_peak_auto.log](https://github.com/mpv-player/mpv/files/13607872/Exodus_HDR_mpv_compute_peak_auto.log)

[Exodus_HDR_mpv_compute_peak_no.log](https://github.com/mpv-player/mpv/files/13607873/Exodus_HDR_mpv_compute_peak_no.log)

### Sample files

https://0x0.st/H3bq.mp4

**Comment by llyyr** [comment]
[Intel DRM bug.](https://gitlab.freedesktop.org/drm/intel/-/issues/9118)

This is fixed in 6.5.7 or newer kernels, so you need to update

**Comment by badgerkin** [comment]
TYVM for the quick response @llyyr! This issue is resolved on my system by updating to the `6.6.4-060604-generic` Ubuntu mainline kernel.
